src/core/plugin/luapi_application.h

```diff
@@ -1976,6 +1978,193 @@ static int applib_openFile(lua_State* L) {
     return 1;
 }
 
+/*
+ * Adds images from the provided paths on the current page on the current layer.
+ *
+ * Global parameters:
+ *  - images table: array of image-parameter-tables
+ *  - allowUndoRedoAction string: Decides how the change gets introduced into the undoRedo action list "individual",
+ *    "grouped" or "none"
+ *
+ * Parameters per image:
+ *  - path string: filepath to the image (required)
+ *  - x number: x-Coordinate where to place the left upper corner of the image (default: 0)
+ *  - y number: y-Coordinate where to place the left upper corner of the image (default: 0)
+ *  scaling options:
+ *  - maxWidth integer: sets the width of the image in pixles. If that is too big for the page the image gets scaled
+ *    down keeping the aspect-ratio provided by maxWidth/masHeight (default: -1)
+ *  - maxHeight integer: sets the height of the image in pixles. If that is too big for the page the image gets scaled
+ *    down keeping the aspect-ratio provided by maxWidth/masHeight (default: -1)
+ *  - aspectRatio boolean: if only one of width and height is given should the image be scaled in the other dimension
```

Comment:
I think when both `maxWidth` and `maxHeight` are given and `aspectRatio` is true (or nil) then the aspect ratio should be preserved.

E.g. for `maxWidth = 500`, `maxHeight = 400`, `real width = 1000`, `realHeight = 900`, `aspectRatio = true` 
we should get a height of 400 and a width of 444 (not 500)

Reply:
I disagree. For me it makes sense, since `maxWidth` only means the `width` should not be larger (and will be as large if nothing speaks against it).
Imagine you want to add multiple images in a grid, where the cells have fixed dimension (with some padding between the cells). I want to have the images as large as possible inside the cells but preserving the aspect ratio. Then I simply specify the `maxWidth` = cell width, `maxHeight` = cell height and `aspectRatio = true` and I don't have to worry, for which images the ratio between width and height is larger or smaller than the ratio between width and height of the cells. So let's say the cells have dimensions `500 x 400`. 
An `1000x900` image with would become `444x400` (it fits into the cell and is as large as possible),
an `600x400` image would become `500x333` (it also fits into the cell and is as large as possible).

Of course if all these three values are given the actual `width` and `height` must be <= `maxWidth` and `maxHeight`, respectively but maximal with the extra condition that the aspect ratio is preserved. You can either compute that by first taking care of `width<=maxWidth` and then `height<=maxHeight` or by comparing the aspect ratios `realWidth/realHeight` and `maxWidth/maxHeight`.

Reply:
See 4750fb47a, should be fixed by now.